### PR TITLE
made nav work

### PR DESCRIPTION
**Pull request by llwor94**

### CODE REVIEW

File: dashboard/client/components/SubMenu/index.tsx [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
-import './styles.scss'
+import React, { Fragment } from 'react'
 
-const SubMenu = ({ spaces }) => {
-  const [selected, setSelected] = useState()
+import './styles.scss'
 
-  const x = t => _ => setSelected(t)
+const SubMenu = ({ spaces, space }) => {
+  const x = (id, name) => _ => Router.push({ pathname: '/home', query: { space: id } })
```
Comment (frankfaustino) [inline dashboard/client/components/SubMenu/index.tsx]: i dig it

File: dashboard/server/src/communityAPI.js [hunk 2]
```diff
@@ -44,7 +46,7 @@ const resolvers = {
     spaces: async () => {
       try {
         let { data } = await server({ method: 'get', url: '/space.json' })
-        return data
+        return data.list.map(item => ({ id: item.id, name: item.name }))
```
Comment (frankfaustino) [inline dashboard/server/src/communityAPI.js]: could shorten it:

```js
return data.list.map(({ id, name }) => ({ id, name }))
```

File: dashboard/client/pages/index.tsx [hunk 3]
```diff
@@ -12,20 +12,32 @@ interface IProps {
...
-    {tickets.map(t => (
-      <Ticket key={t.title} ticket={t} />
-    ))}
-  </Layout>
-)
+const HomePage: NextStatelessComponent<IProps> = ({ spaces, tickets: initalTickets, query }) => {
+  const [tickets, setTickets] = useState(initalTickets)
+
+  useEffect(() => {
+    const fetchTickets = async () => {
```
Comment (frankfaustino) [inline dashboard/client/pages/index.tsx]: do we rly need this outer function `fetchTickets`?

File: dashboard/client/pages/index.tsx [hunk 4]
```diff
@@ -12,20 +12,32 @@ interface IProps {
...
+      const { data } = await getTickets(query.space)
```
Comment (frankfaustino) [inline dashboard/client/pages/index.tsx]: we should be makin sure object properties even exist:

```
if (query && query.space) {
  const { data } = await getTickets(query.space)

  if (data && data.tickets) {
    setTickets(data.tickets)
  }
}
```

**Comment by frankfaustino** [comment]
good stuff 😊

File: dashboard/client/pages/index.tsx (hunk 3 above)
Comment (llwor94) [reply dashboard/client/pages/index.tsx]: you cant put an async directly on useEffect bc useEffect should either return a cleanup function or nothin according to the docs, it'll give u a lil warning

File: dashboard/client/pages/index.tsx (hunk 4 above)
Comment (llwor94) [reply dashboard/client/pages/index.tsx]: query always exists w/ nextjs, if there's no query it's just an empty object

File: dashboard/server/src/communityAPI.js (hunk 2 above)
Comment (llwor94) [reply dashboard/server/src/communityAPI.js]: 👌

File: dashboard/client/pages/index.tsx (hunk 3 above)
Comment (frankfaustino) [reply dashboard/client/pages/index.tsx]: good to know!

**Pull request merged**